app/models/tag.rb

```diff
@@ -995,7 +996,7 @@ def syn_string=(tag_string)
     if new_merger && new_merger == self
       self.errors.add(:base, tag_string + " is considered the same as " + self.name + " by the database.")
     elsif new_merger && !new_merger.canonical?
-      self.errors.add(:base, '<a href="/tags/' + new_merger.to_param + '/edit">' + new_merger.name + '</a> is not a canonical tag. Please make it canonical before adding synonyms to it.')
+      self.errors.add(:base, "<a href=\"#{Rails.application.routes.url_helpers.edit_tag_path(new_merger)}\">#{new_merger.name}</a> is not a canonical tag. Please make it canonical before adding synonyms to it.")
```

Comment:
Since you've added `include Rails.application.routes.url_helpers` at the top, just `edit_tag_path(new_merger)` should work here.